step-templates/newrelic-complete-deployment.json

```diff
@@ -3,9 +3,9 @@
--------\n## Configuration\n## --------------------------------------------------------------------------------------\nValidate-Parameter $apiKey -parameterName \"Api Key\"\n\nif (!$appId) {\n    Write-Host \"NewRelic Deploy - AppId has not been set yet. Skipping call to API.\"\n    exit 0\n}\n\nif ($appId -eq 0) {\n    Write-Host \"NewRelic Deploy - AppId has been set to zero. Skipping call to API.\"\n    exit 0\n}\n\nWrite-Host \"NewRelic Deploy - Notify deployment by $user - App $appId - Rev $revision\"\n\n        \n# https://rpm.newrelic.com/api/explore/application_deployments/create?application_id=1127348\n$deployment = New-Object -TypeName PSObject\n$deployment | Add-Member -MemberType NoteProperty -Name \"user\" -Value $user\n$deployment | Add-Member -MemberType NoteProperty -Name \"revision\" -Value $releaseNumber\n$deployment | Add-Member -MemberType NoteProperty -Name \"changelog\" -Value $releaseNotes\n$deployment | Add-Member -MemberType NoteProperty -Name \"description\" -Value \"Octopus deployment of $projectName to $machineName. ($deploymentLink)\"\n\n$deploymentContainer = New-Object -TypeName PSObject\n$deploymentContainer | Add-Member -MemberType NoteProperty -Name \"deployment\" -Value $deployment\n\n$post = $deploymentContainer | ConvertTo-Json\nWrite-Debug $post\n\n# in production, we need to\n#Create a URI instance since the HttpWebRequest.Create Method will escape the URL by default.\n$URL = \"https://api.newrelic.com/v2/applications/$appId/deployments.json\"\n$URI = New-Object System.Uri($URL,$true)\n\n#Create a request object using the URI\n$request = [System.Net.HttpWebRequest]::Create($URI)\n$request.Method = \"POST\"\n$request.Headers.Add(\"X-Api-Key\",\"$apiKey\");\n$request.ContentType = \"application/json\"\n\n#Build up a nice User Agent\n$request.UserAgent = $(\n    \"{0} (PowerShell {1}; .NET CLR {2}; {3})\" -f $UserAgent, \n    $(if($Host.Version){$Host.Version}else{\"1.0\"}),\n    [Environment]::Version,\n    [Environment]::OSVersion.ToString().Replace(\"Microsoft Windows \", \"Win\")\n)\n\ntry {\n    Write-Host \"Posting data to $URL\"\n    #Create a new stream writer to write the xml to the request stream.\n    $stream = New-Object IO.StreamWriter $request.GetRequestStream()\n    $stream.AutoFlush = $True\n    $PostStr = [System.Text.Encoding]::UTF8.GetBytes($Post)\n    $stream.Write($PostStr, 0,$PostStr.length)\n    $stream.Close()\n \n    #Make the request and get the response\n    $response = $request.GetResponse()    \n    $response.Close()\n\n    if ([int]$response.StatusCode -eq 201) {\n        Write-Host \"NewRelic Deploy - API called succeeded - HTTP $($response.StatusCode).\"\n    } else {\n        Write-Host \"NewRelic Deploy - API called failed - HTTP $($response.StatusCode).\"\n        exit 1\n    }\n} catch [System.Net.WebException] {\n    $ErrorMessage = $_.Exception.Message\n    $res = $_.Exception.Response\n    Write-Host \"NewRelic Deploy - API called failed - $ErrorMessage\" \n    exit 1\n}",
```

Comment:
Hi @naile and thanks for this contribution!

With the user parameter now being optional, this line needs a small update:

```powershell
Write-Host "NewRelic Deploy - Notify deployment by $user - App $appId - Rev $revision"
```

Can you update it to work with or without the $user value? If not, I can make the change for you.

Reply:
Ah missed that one. Sure thing. I'm not too well versed in Powershell, would something like this be acceptable?
```PowerShell
Write-Host "NewRelic Deploy - Notify deployment" $(("by user $user", "")[!$user]) "- App $appId - Rev $revision"
```